### PR TITLE
parseDateTime[InJodaSyntax](): Require format argument

### PR DESCRIPTION
**Pull request by rschu1ze**
Function parseDateTime\[InJodaSyntax\]() previously accepted an (undocumented) default format string. The problem was that
1. there is no "natural" default format
2. MySQL str_to_date() (= the motivation for parseDateTime\[InJodaSyntax\]()) also provides no default format

--> get rid of the default format

@taiyang-li 

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Functions/parseDateTime.cpp [hunk 1]
```diff
@@ -494,14 +494,14 @@ namespace
                     arguments[0].type->getName(),
                     getName());
 
-            if (arguments.size() > 1 && !isString(arguments[1].type))
+            if (!isString(arguments[1].type))
```
Comment (davenger) [inline src/Functions/parseDateTime.cpp:497]: Maybe use validateFunctionArgumentTypes helper? Like here:
https://github.com/ClickHouse/ClickHouse/blob/master/src/Functions/regexpExtract.cpp#L49-L57

Comment (rschu1ze) [reply src/Functions/parseDateTime.cpp:497]: Tnx, didn't know about this class. Sounds cool, will use it.

**Review by davenger — APPROVED** [review]

**Comment by seandhaynes** [comment]
@rschu1ze Hey, I found this commit as part of upgrading to a v24.3 release.

This breaks the backwards compatibility integration test for [string functions](https://github.com/ClickHouse/ClickHouse/blob/master/tests/integration/test_backward_compatibility/test_functions.py).

It looks like it's currently passing in upstream's master branch as the test case uses `CLICKHOUSE_CI_MIN_TESTED_VERSION` which is currently set to v22.3, which doesn't have these functions, so the tests are skipped.

I completely agree that a format makes sense, so thank you for this! That said, do you think it's possible to default to the old behaviour if a format isn't specified, and make the format an optional argument?

I say this as there may have already been clients that don't expect an exception when just providing a single string (including us - I haven't finished checking all clusters yet).

**Comment by rschu1ze** [comment]
@seandhaynes Thanks for the suggestion, done! https://github.com/ClickHouse/ClickHouse/pull/67399

**Comment by seandhaynes** [comment]
Thank you very much, @rschu1ze !

**Pull request merged**